### ISSUE TITLE
apply-changes error message when errand not in list of selected products

### ISSUE THREAD

**Issue by patrick-huber-pivotal** [pivotal-cf/om#431]
When applying changes with selective apply changes you have the ability to specify a config yml file with the errands listed. If you happen to include an errand name that isn't in the list of selected products, you get the following error message seen here:

```
failed to fetch product GUID for product: %s
```

Where %s is the name of the product. 

https://github.com/pivotal-cf/om/blob/39ed12e0f07df883fa0e0cd4921db0d816a3dafb/api/installations_service.go#L105

What isn't clear to the user is that the product actually does exist and does have a guid, but they didn't specify the product name in the selected list of products so the guid was actually filtered out. 

Perhaps change the error message here to state that the errand exists but they didn't specify the product name in the selective apply changes.

**Comment by kcboyle** [comment]
makes sense. If we created a check for the optional errands to require the product name, would that be sufficient feedback? @patrick-huber-pivotal

**Comment by patrick-huber-pivotal** [comment]
As long as the user is informed of why the validation is failing. We had to
dig through the source to find out that our product list was causing the
listed errands to fail validation.

**Comment by kcboyle** [comment]
@patrick-huber-pivotal can you verify for me whether you are using --product-name when you run this command? or are you running just `apply-changes` with the `--config`?  

I just ran a test `om --env env.yml apply-changes -c errands.yml` and got no failures. Is this only an issue during a selective deploy?

**Comment by kcboyle** [comment]
@patrick-huber-pivotal  could you follow up on this for us, please?

**Comment by patrick-huber-pivotal** [comment]
We were using the ProductNames list using the config file: 

https://github.com/pivotal-cf/om/blob/4de8213494dd7038d3dd7a5f24274433fcaa8d55/commands/apply_changes.go#L27

It was a selective apply changes where one of the errands in the list is not an actual errand.

**Comment by kcboyle** [comment]
@patrick-huber-pivotal this was released in om 5.0.0
